packages/plugin-legacy/src/index.ts

```diff
@@ -339,6 +333,10 @@ function viteLegacyPlugin(options: Options = {}): Plugin[] {
       isDebug &&
         console.log(`[@vitejs/plugin-legacy] modernTargets:`, modernTargets)
 
+      if (!genLegacy || config.build.ssr) {
+        return
+      }
```

Comment:
I think perhaps we can only have `modernTargets` above this if condition? If `genLegacy` is true, I don't think we have to compute the `targets` variable.